### PR TITLE
Always call onPostDeactivate even if focus not returned

### PR DESCRIPTION
**Pull request by stefcameron**
This is consistent with always calling onPostActivate after
onActivate.

I enhanced the Animated Trigger demo to show that
onPostDeactivate is always called even if `returnFocus`
is not set.

And I added visible (i.e. on-screen) messages for
the Animated Dialog and Animated Trigger demos
instead of logging to the console. This way, I could
add a test for the Tigger where focus is not returned
but onPostDeactivate is still called.

<details>
<summary>PR Checklist</summary>
<br/>

__Please leave this checklist in your PR.__

- Source changes maintain stated browser compatibility.
- Includes updated `/docs/demo-bundle.js` if source or docs code was changed.
- Issue being fixed is referenced.
- Unit test coverage added/updated.
- E2E test coverage added/updated.
- Typings added/updated.
- README updated (API changes, instructions, etc.).
- Changes to dependencies explained.
- Changeset added (run `yarn changeset` locally to add one, and follow the prompts).
  - EXCEPTION: A Changeset is not required if the change does not affect any of the source files that produce the package bundle. For example, tooling changes, test updates, or a new dev-only dependency to run tests more efficiently should not have a Changeset since it will not affect package consumers.

</details>

### CODE REVIEW

File: docs/demo-bundle.js [hunk 1]
```diff
@@ -564,15 +564,15 @@ var createFocusTrap = function createFocusTrap(elements, userOptions) {
       var returnFocus = getOption(deactivateOptions, 'returnFocus', 'returnFocusOnDeactivate');
 
       var finishDeactivation = function finishDeactivation() {
-        if (returnFocus) {
```
Comment (stefcameron) [inline docs/demo-bundle.js:567]: Ignore this file, it's generated from `yarn demo-bundle`

**Comment by stefcameron** [comment]
@Dan503 Please check this out when you have a moment.

**Comment by Dan503** [comment]
I haven't forgotten about this. I just haven't had a chance to review yet.

**Review by Dan503 — COMMENTED** [review]
I've made some code suggestions in this PR (I don't have permission to push to your PR)

https://github.com/focus-trap/focus-trap/pull/434

**Review by Dan503 — CHANGES_REQUESTED** [review]
Sorry, I was supposed to check the "Request changes" radio button.

File: docs/index.html [hunk 2]
```diff
@@ -47,14 +47,17 @@ <h2 id="default-heading">default behavior</h2>
     <div id="demo-animated-dialog">
       <h2 id="animated-dialog-heading">animated dialog</h2>
       <p>
-        This focus trap will fade in when activated and fade out when deactivated. It logs a message to the
-        console when the trap has finished activating. Note that the fade animation needs to be written yourself,
-        it is not built into the library.
+        This focus trap will fade in when activated and fade out when deactivated. Once the trap has been activated
+        (i.e. <code>onPostActivate</code>), a &quot;trap activated&quot; message appears next to the Activate button.
```
[the remark this answers is not in the document]
Comment (stefcameron) [reply docs/index.html:51]: Force of habit from React always complaining about using characters like the apostrophe instead of `&apos;` for example.

**Review by Dan503 — APPROVED** [review]

**Pull request merged**